From ef3afb65b36dc8e96c0b881551a20f64be4196d7 Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Thu, 3 Jun 2021 16:03:48 +0800
Subject: [PATCH] Refine the file_contexts of block devices

There is no need to use Regular expression to
denote the block devices' path, it's very inefficient.

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 boot-arch/generic/file_contexts | 40 ++++++++++++++++-----------------
 config-partition/file_contexts  |  2 +-
 2 files changed, 21 insertions(+), 21 deletions(-)

diff --git a/boot-arch/generic/file_contexts b/boot-arch/generic/file_contexts
index c8fc1f3..4984f6c 100644
--- a/boot-arch/generic/file_contexts
+++ b/boot-arch/generic/file_contexts
@@ -1,23 +1,23 @@
 #
 # Block Devices
 #
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/boot(_(a|b))?	u:object_r:boot_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/bootloader(_(a|b))?	u:object_r:boot_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/multiboot(_(a|b))?	u:object_r:boot_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/system(_(a|b))?	u:object_r:system_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/vendor(_(a|b))?	u:object_r:vendor_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/product(_(a|b))?	u:object_r:product_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/odm(_(a|b))?	u:object_r:odm_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/vbmeta(_(a|b))?	u:object_r:boot_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/tos(_(a|b))?		u:object_r:boot_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/persistent		u:object_r:frp_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/recovery		u:object_r:recovery_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/metadata		u:object_r:metadata_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/cache		u:object_r:cache_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/data			u:object_r:userdata_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/userdata			u:object_r:userdata_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/misc			u:object_r:misc_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/teedata		u:object_r:tee_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/acpi(_(a|b))?		u:object_r:acpi_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/acpio(_(a|b))?		u:object_r:acpio_block_device:s0
-/dev/block/(pci|platform|vbd)(/.*)?/.*/by-name/super 		u:object_r:super_block_device:s0
+/dev/block/by-name/boot(_(a|b))?	u:object_r:boot_block_device:s0
+/dev/block/by-name/bootloader(_(a|b))?	u:object_r:boot_block_device:s0
+/dev/block/by-name/multiboot(_(a|b))?	u:object_r:boot_block_device:s0
+/dev/block/by-name/system(_(a|b))?	u:object_r:system_block_device:s0
+/dev/block/by-name/vendor(_(a|b))?	u:object_r:vendor_block_device:s0
+/dev/block/by-name/product(_(a|b))?	u:object_r:product_block_device:s0
+/dev/block/by-name/odm(_(a|b))?	u:object_r:odm_block_device:s0
+/dev/block/by-name/vbmeta(_(a|b))?	u:object_r:boot_block_device:s0
+/dev/block/by-name/tos(_(a|b))?		u:object_r:boot_block_device:s0
+/dev/block/by-name/persistent		u:object_r:frp_block_device:s0
+/dev/block/by-name/recovery		u:object_r:recovery_block_device:s0
+/dev/block/by-name/metadata		u:object_r:metadata_block_device:s0
+/dev/block/by-name/cache		u:object_r:cache_block_device:s0
+/dev/block/by-name/data			u:object_r:userdata_block_device:s0
+/dev/block/by-name/userdata			u:object_r:userdata_block_device:s0
+/dev/block/by-name/misc			u:object_r:misc_block_device:s0
+/dev/block/by-name/teedata		u:object_r:tee_device:s0
+/dev/block/by-name/acpi(_(a|b))?		u:object_r:acpi_block_device:s0
+/dev/block/by-name/acpio(_(a|b))?		u:object_r:acpio_block_device:s0
+/dev/block/by-name/super 		u:object_r:super_block_device:s0
diff --git a/config-partition/file_contexts b/config-partition/file_contexts
index 8f532c7..7798182 100644
--- a/config-partition/file_contexts
+++ b/config-partition/file_contexts
@@ -1,4 +1,4 @@
 # Config partition
 /oem_config(/.*)?               u:object_r:config_file:s0
 
-/dev/block/(pci|platform)(/.*)?/.*/by-name/config  u:object_r:config_block_device:s0
+/dev/block/by-name/config  u:object_r:config_block_device:s0
-- 
2.25.1

